stdlib/asm/mem.masm

```diff
@@ -0,0 +1,41 @@
+# ===== MEMORY FUNCTIONS ==========================================================================
+
+#! memcopy(read_ptr: address, write_ptr: address, words: felt)
+#!
+#! Copies `words` words from `read_ptr` to `write_ptr`.
+#!
+#! Stack transition looks as follows:
+#! [read_ptr, write_ptr, words, ...] -> [...]
+export.memcopy
+  # add counter
+  # stack: [read_ptr, write_ptr, count, words, ...]
+  push.0 movdn.2
+
+  # Pad the stack because mem_load overwrites it
+  # stack: [PAD, read_ptr, write_ptr, count, words, ...]
+  padw
+
+  # LOOP: [PAD, read_ptr, write_ptr, count, words, ...]
+  # while(words>0) (17 cycles)
+  dup.7 dup.7 neq while.true
+    # perform read (2 cycles)
+    # stack: [A, read_ptr, write_ptr, count, words, ...]
+    dup.4 mem_loadw
+
+    # peform write (3 cycles)
+    # stack: [A, read_ptr, write_ptr, count, words, ...]
+    dup.5 mem_storew
```

Comment:
why is this 3 cycles? Shouldn't it be 2?

Reply:
ah, I looked the cycle count of `mem_store` instead of `mem_storew`. Thanks!